miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -478,6 +530,33 @@ private void setIdentity(DetailedSample descendant) throws IOException, MisoNami
     child.setParent(identity);
   }
 
+  /**
+   * Finds the identity of a sample by climbing the hierarchy
+   * 
+   * @param sample
+   * @return identity attached to sample or null if no identity is attached
+   * @throws IOException
+   */
+  private SampleIdentity getIdentity(DetailedSample sample) throws IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)